File: com.bosch.fsp.master_of_plow.gen/src/com/bosch/nevonex/fcal/impl/GPS_NMEA0183_T_R.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.fcal.impl;

import com.bosch.nevonex.fcal.IGPS_NMEA0183_T_R;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * @generated
 */
public class GPS_NMEA0183_T_R extends EObjectImpl implements IGPS_NMEA0183_T_R {
	protected String uTC_ms = null;
	protected double latitude = 0.0;
	protected double longitude = 0.0;
	protected int gPS_Quality_Indicator = 0;
	protected int no_Satellites = 0;
	protected float altitude = 0.0f;
	protected float geoidal_Separation = 0.0f;
	protected float dGPS_Age = 0.0f;
	protected int differential_Ref_Station = 0;
	protected String selection_Mode = null;
	protected int fix_Type = 0;
	protected float pDOP = 0.0f;
	protected float hDOP = 0.0f;
	protected float vDOP = 0.0f;
	protected int status = 0;
	protected float speed = 0.0f;
	protected float course = 0.0f;
	protected float magnetic_variation = 0.0f;
	protected String fAA_Mode_Indicator = null;
	protected float heading = 0.0f;
	protected int headingTrue = 0;

	protected GPS_NMEA0183_T_R() {
		super();
	}

	@Override
	protected EClass eStaticClass() {
		return FcalPackage.Literals.GPS_NMEA0183_T_R;
	}

	public String getUTC_ms() {
		return uTC_ms;
	}

	public void setUTC_ms(String newUTC_ms) {
		uTC_ms = newUTC_ms;
	}

	public double getLatitude() {
		return latitude;
	}

	public void setLatitude(double newLatitude) {
		latitude = newLatitude;
	}

	public double getLongitude() {
		return longitude;
	}

	public void setLongitude(double newLongitude) {
		longitude = newLongitude;
	}

	public int getGPS_Quality_Indicator() {
		return gPS_Quality_Indicator;
	}

	public void setGPS_Quality_Indicator(int newGPS_Quality_Indicator) {
		gPS_Quality_Indicator = newGPS_Quality_Indicator;
	}

	public int getNo_Satellites() {
		return no_Satellites;
	}

	public void setNo_Satellites(int newNo_Satellites) {
		no_Satellites = newNo_Satellites;
	}

	public float getAltitude() {
		return altitude;
	}

	public void setAltitude(float newAltitude) {
		altitude = newAltitude;
	}

	public float getGeoidal_Separation() {
		return geoidal_Separation;
	}

	public void setGeoidal_Separation(float newGeoidal_Separation) {
		geoidal_Separation = newGeoidal_Separation;
	}

	public float getDGPS_Age() {
		return dGPS_Age;
	}

	public void setDGPS_Age(float newDGPS_Age) {
		dGPS_Age = newDGPS_Age;
	}

	public int getDifferential_Ref_Station() {
		return differential_Ref_Station;
	}

	public void setDifferential_Ref_Station(int newDifferential_Ref_Station) {
		differential_Ref_Station = newDifferential_Ref_Station;
	}

	public String getSelection_Mode() {
		return selection_Mode;
	}

	public void setSelection_Mode(String newSelection_Mode) {
		selection_Mode = newSelection_Mode;
	}

	public int getFix_Type() {
		return fix_Type;
	}

	public void setFix_Type(int newFix_Type) {
		fix_Type = newFix_Type;
	}

	public float getPDOP() {
		return pDOP;
	}

	public void setPDOP(float newPDOP) {
		pDOP = newPDOP;
	}

	public float getHDOP() {
		return hDOP;
	}

	public void setHDOP(float newHDOP) {
		hDOP = newHDOP;
	}

	public float getVDOP() {
		return vDOP;
	}

	public void setVDOP(float newVDOP) {
		vDOP = newVDOP;
	}

	public int getStatus() {
		return status;
	}

	public void setStatus(int newStatus) {
		status = newStatus;
	}

	public float getSpeed() {
		return speed;
	}

	public void setSpeed(float newSpeed) {
		speed = newSpeed;
	}

	public float getCourse() {
		return course;
	}

	public void setCourse(float newCourse) {
		course = newCourse;
	}

	public float getMagnetic_variation() {
		return magnetic_variation;
	}

	public void setMagnetic_variation(float newMagnetic_variation) {
		magnetic_variation = newMagnetic_variation;
	}

	public String getFAA_Mode_Indicator() {
		return fAA_Mode_Indicator;
	}

	public void setFAA_Mode_Indicator(String newFAA_Mode_Indicator) {
		fAA_Mode_Indicator = newFAA_Mode_Indicator;
	}

	public float getHeading() {
		return heading;
	}

	public void setHeading(float newHeading) {
		heading = newHeading;
	}

	public int getHeadingTrue() {
		return headingTrue;
	}

	public void setHeadingTrue(int newHeadingTrue) {
		headingTrue = newHeadingTrue;
	}

	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case FcalPackage.GPS_NMEA0183_T_R__UTC_MS:
			return getUTC_ms();
		case FcalPackage.GPS_NMEA0183_T_R__LATITUDE:
			return getLatitude();
		case FcalPackage.GPS_NMEA0183_T_R__LONGITUDE:
			return getLongitude();
		case FcalPackage.GPS_NMEA0183_T_R__GPS_QUALITY_INDICATOR:
			return getGPS_Quality_Indicator();
		case FcalPackage.GPS_NMEA0183_T_R__NO_SATELLITES:
			return getNo_Satellites();
		case FcalPackage.GPS_NMEA0183_T_R__ALTITUDE:
			return getAltitude();
		case FcalPackage.GPS_NMEA0183_T_R__GEOIDAL_SEPARATION:
			return getGeoidal_Separation();
		case FcalPackage.GPS_NMEA0183_T_R__DGPS_AGE:
			return getDGPS_Age();
		case FcalPackage.GPS_NMEA0183_T_R__DIFFERENTIAL_REF_STATION:
			return getDifferential_Ref_Station();
		case FcalPackage.GPS_NMEA0183_T_R__SELECTION_MODE:
			return getSelection_Mode();
		case FcalPackage.GPS_NMEA0183_T_R__FIX_TYPE:
			return getFix_Type();
		case FcalPackage.GPS_NMEA0183_T_R__PDOP:
			return getPDOP();
		case FcalPackage.GPS_NMEA0183_T_R__HDOP:
			return getHDOP();
		case FcalPackage.GPS_NMEA0183_T_R__VDOP:
			return getVDOP();
		case FcalPackage.GPS_NMEA0183_T_R__STATUS:
			return getStatus();
		case FcalPackage.GPS_NMEA0183_T_R__SPEED:
			return getSpeed();
		case FcalPackage.GPS_NMEA0183_T_R__COURSE:
			return getCourse();
		case FcalPackage.GPS_NMEA0183_T_R__MAGNETIC_VARIATION:
			return getMagnetic_variation();
		case FcalPackage.GPS_NMEA0183_T_R__FAA_MODE_INDICATOR:
			return getFAA_Mode_Indicator();
		case FcalPackage.GPS_NMEA0183_T_R__HEADING:
			return getHeading();
		case FcalPackage.GPS_NMEA0183_T_R__HEADING_TRUE:
			return getHeadingTrue();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case FcalPackage.GPS_NMEA0183_T_R__UTC_MS:
			setUTC_ms((String) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__LATITUDE:
			setLatitude((Double) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__LONGITUDE:
			setLongitude((Double) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__GPS_QUALITY_INDICATOR:
			setGPS_Quality_Indicator((Integer) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__NO_SATELLITES:
			setNo_Satellites((Integer) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__ALTITUDE:
			setAltitude((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__GEOIDAL_SEPARATION:
			setGeoidal_Separation((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__DGPS_AGE:
			setDGPS_Age((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__DIFFERENTIAL_REF_STATION:
			setDifferential_Ref_Station((Integer) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__SELECTION_MODE:
			setSelection_Mode((String) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__FIX_TYPE:
			setFix_Type((Integer) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__PDOP:
			setPDOP((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__HDOP:
			setHDOP((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__VDOP:
			setVDOP((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__STATUS:
			setStatus((Integer) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__SPEED:
			setSpeed((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__COURSE:
			setCourse((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__MAGNETIC_VARIATION:
			setMagnetic_variation((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__FAA_MODE_INDICATOR:
			setFAA_Mode_Indicator((String) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__HEADING:
			setHeading((Float) newValue);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__HEADING_TRUE:
			setHeadingTrue((Integer) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case FcalPackage.GPS_NMEA0183_T_R__UTC_MS:
			setUTC_ms(null);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__LATITUDE:
			setLatitude(0.0);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__LONGITUDE:
			setLongitude(0.0);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__GPS_QUALITY_INDICATOR:
			setGPS_Quality_Indicator(0);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__NO_SATELLITES:
			setNo_Satellites(0);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__ALTITUDE:
			setAltitude(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__GEOIDAL_SEPARATION:
			setGeoidal_Separation(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__DGPS_AGE:
			setDGPS_Age(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__DIFFERENTIAL_REF_STATION:
			setDifferential_Ref_Station(0);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__SELECTION_MODE:
			setSelection_Mode(null);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__FIX_TYPE:
			setFix_Type(0);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__PDOP:
			setPDOP(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__HDOP:
			setHDOP(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__VDOP:
			setVDOP(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__STATUS:
			setStatus(0);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__SPEED:
			setSpeed(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__COURSE:
			setCourse(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__MAGNETIC_VARIATION:
			setMagnetic_variation(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__FAA_MODE_INDICATOR:
			setFAA_Mode_Indicator(null);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__HEADING:
			setHeading(0.0f);
			return;
		case FcalPackage.GPS_NMEA0183_T_R__HEADING_TRUE:
			setHeadingTrue(0);
			return;
		}
		super.eUnset(featureID);
	}

	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case FcalPackage.GPS_NMEA0183_T_R__UTC_MS:
			return uTC_ms != null;
		case FcalPackage.GPS_NMEA0183_T_R__LATITUDE:
			return latitude != 0.0;
		case FcalPackage.GPS_NMEA0183_T_R__LONGITUDE:
			return longitude != 0.0;
		case FcalPackage.GPS_NMEA0183_T_R__GPS_QUALITY_INDICATOR:
			return gPS_Quality_Indicator != 0;
		case FcalPackage.GPS_NMEA0183_T_R__NO_SATELLITES:
			return no_Satellites != 0;
		case FcalPackage.GPS_NMEA0183_T_R__ALTITUDE:
			return altitude != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__GEOIDAL_SEPARATION:
			return geoidal_Separation != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__DGPS_AGE:
			return dGPS_Age != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__DIFFERENTIAL_REF_STATION:
			return differential_Ref_Station != 0;
		case FcalPackage.GPS_NMEA0183_T_R__SELECTION_MODE:
			return selection_Mode != null;
		case FcalPackage.GPS_NMEA0183_T_R__FIX_TYPE:
			return fix_Type != 0;
		case FcalPackage.GPS_NMEA0183_T_R__PDOP:
			return pDOP != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__HDOP:
			return hDOP != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__VDOP:
			return vDOP != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__STATUS:
			return status != 0;
		case FcalPackage.GPS_NMEA0183_T_R__SPEED:
			return speed != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__COURSE:
			return course != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__MAGNETIC_VARIATION:
			return magnetic_variation != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__FAA_MODE_INDICATOR:
			return fAA_Mode_Indicator != null;
		case FcalPackage.GPS_NMEA0183_T_R__HEADING:
			return heading != 0.0f;
		case FcalPackage.GPS_NMEA0183_T_R__HEADING_TRUE:
			return headingTrue != 0;
		}
		return super.eIsSet(featureID);
	}
} //GPS_NMEA0183_T_R

File: master_of_plow_CPP_SDK/src-gen/nevonex/fcal/GPS_NMEA0183_T_R.cpp
/*
 * nevonex/fcal/GPS_NMEA0183_T_R.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "GPS_NMEA0183_T_R.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/fcal/FcalPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <algorithm>
#include <nevonex/fcb/SubscriberEnum.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

#if BOOST_VERSION >= 106501
#define BOOST_STACKTRACE_HEADERS_FOUND
#include <boost/stacktrace.hpp>
#endif

/*PROTECTED REGION ID(GPS_NMEA0183_T_R.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::fcal;
using namespace ::nevonex::log;

// Default constructor
GPS_NMEA0183_T_R::GPS_NMEA0183_T_R() : m_uTC_ms(""), m_latitude(0.0), m_longitude(0.0), m_gPS_Quality_Indicator(0), m_no_Satellites(0), m_altitude(0.0f), m_geoidal_Separation(0.0f), m_dGPS_Age(0.0f), m_differential_Ref_Station(0), m_selection_Mode(""), m_fix_Type(0), m_pDOP(0.0f), m_hDOP(0.0f), m_vDOP(0.0f), m_status(0), m_speed(0.0f), m_course(0.0f), m_magnetic_variation(0.0f), m_fAA_Mode_Indicator(""), m_heading(0.0f), m_headingTrue(0)
{

    /*PROTECTED REGION ID(GPS_NMEA0183_T_R__GPS_NMEA0183_T_R) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

GPS_NMEA0183_T_R::~GPS_NMEA0183_T_R()
{

    /*PROTECTED REGION ID(GPS_NMEA0183_T_R__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

// Attributes

::ecore::EString GPS_NMEA0183_T_R::getUTC_ms() const
{
    return m_uTC_ms;
}

void GPS_NMEA0183_T_R::setUTC_ms(::ecore::EString _uTC_ms)
{

    ::ecore::EString _old_uTC_ms = m_uTC_ms;
    m_uTC_ms = _uTC_ms;

}

::ecore::EDouble GPS_NMEA0183_T_R::getLatitude() const
{
    return m_latitude;
}

void GPS_NMEA0183_T_R::setLatitude(::ecore::EDouble _latitude)
{

    ::ecore::EDouble _old_latitude = m_latitude;
    m_latitude = _latitude;

}

::ecore::EDouble GPS_NMEA0183_T_R::getLongitude() const
{
    return m_longitude;
}

void GPS_NMEA0183_T_R::setLongitude(::ecore::EDouble _longitude)
{

    ::ecore::EDouble _old_longitude = m_longitude;
    m_longitude = _longitude;

}

::ecore::EInt GPS_NMEA0183_T_R::getGPS_Quality_Indicator() const
{
    return m_gPS_Quality_Indicator;
}

void GPS_NMEA0183_T_R::setGPS_Quality_Indicator(::ecore::EInt _gPS_Quality_Indicator)
{

    ::ecore::EInt _old_gPS_Quality_Indicator = m_gPS_Quality_Indicator;
    m_gPS_Quality_Indicator = _gPS_Quality_Indicator;

}

::ecore::EInt GPS_NMEA0183_T_R::getNo_Satellites() const
{
    return m_no_Satellites;
}

void GPS_NMEA0183_T_R::setNo_Satellites(::ecore::EInt _no_Satellites)
{

    ::ecore::EInt _old_no_Satellites = m_no_Satellites;
    m_no_Satellites = _no_Satellites;

}

::ecore::EFloat GPS_NMEA0183_T_R::getAltitude() const
{
    return m_altitude;
}

void GPS_NMEA0183_T_R::setAltitude(::ecore::EFloat _altitude)
{

    ::ecore::EFloat _old_altitude = m_altitude;
    m_altitude = _altitude;

}

::ecore::EFloat GPS_NMEA0183_T_R::getGeoidal_Separation() const
{
    return m_geoidal_Separation;
}

void GPS_NMEA0183_T_R::setGeoidal_Separation(::ecore::EFloat _geoidal_Separation)
{

    ::ecore::EFloat _old_geoidal_Separation = m_geoidal_Separation;
    m_geoidal_Separation = _geoidal_Separation;

}

::ecore::EFloat GPS_NMEA0183_T_R::getDGPS_Age() const
{
    return m_dGPS_Age;
}

void GPS_NMEA0183_T_R::setDGPS_Age(::ecore::EFloat _dGPS_Age)
{

    ::ecore::EFloat _old_dGPS_Age = m_dGPS_Age;
    m_dGPS_Age = _dGPS_Age;

}

::ecore::EInt GPS_NMEA0183_T_R::getDifferential_Ref_Station() const
{
    return m_differential_Ref_Station;
}

void GPS_NMEA0183_T_R::setDifferential_Ref_Station(::ecore::EInt _differential_Ref_Station)
{

    ::ecore::EInt _old_differential_Ref_Station = m_differential_Ref_Station;
    m_differential_Ref_Station = _differential_Ref_Station;

}

::ecore::EString GPS_NMEA0183_T_R::getSelection_Mode() const
{
    return m_selection_Mode;
}

void GPS_NMEA0183_T_R::setSelection_Mode(::ecore::EString _selection_Mode)
{

    ::ecore::EString _old_selection_Mode = m_selection_Mode;
    m_selection_Mode = _selection_Mode;

}

::ecore::EInt GPS_NMEA0183_T_R::getFix_Type() const
{
    return m_fix_Type;
}

void GPS_NMEA0183_T_R::setFix_Type(::ecore::EInt _fix_Type)
{

    ::ecore::EInt _old_fix_Type = m_fix_Type;
    m_fix_Type = _fix_Type;

}

::ecore::EFloat GPS_NMEA0183_T_R::getPDOP() const
{
    return m_pDOP;
}

void GPS_NMEA0183_T_R::setPDOP(::ecore::EFloat _pDOP)
{

    ::ecore::EFloat _old_pDOP = m_pDOP;
    m_pDOP = _pDOP;

}

::ecore::EFloat GPS_NMEA0183_T_R::getHDOP() const
{
    return m_hDOP;
}

void GPS_NMEA0183_T_R::setHDOP(::ecore::EFloat _hDOP)
{

    ::ecore::EFloat _old_hDOP = m_hDOP;
    m_hDOP = _hDOP;

}

::ecore::EFloat GPS_NMEA0183_T_R::getVDOP() const
{
    return m_vDOP;
}

void GPS_NMEA0183_T_R::setVDOP(::ecore::EFloat _vDOP)
{

    ::ecore::EFloat _old_vDOP = m_vDOP;
    m_vDOP = _vDOP;

}

::ecore::EInt GPS_NMEA0183_T_R::getStatus() const
{
    return m_status;
}

void GPS_NMEA0183_T_R::setStatus(::ecore::EInt _status)
{

    ::ecore::EInt _old_status = m_status;
    m_status = _status;

}

::ecore::EFloat GPS_NMEA0183_T_R::getSpeed() const
{
    return m_speed;
}

void GPS_NMEA0183_T_R::setSpeed(::ecore::EFloat _speed)
{

    ::ecore::EFloat _old_speed = m_speed;
    m_speed = _speed;

}

::ecore::EFloat GPS_NMEA0183_T_R::getCourse() const
{
    return m_course;
}

void GPS_NMEA0183_T_R::setCourse(::ecore::EFloat _course)
{

    ::ecore::EFloat _old_course = m_course;
    m_course = _course;

}

::ecore::EFloat GPS_NMEA0183_T_R::getMagnetic_variation() const
{
    return m_magnetic_variation;
}

void GPS_NMEA0183_T_R::setMagnetic_variation(::ecore::EFloat _magnetic_variation)
{

    ::ecore::EFloat _old_magnetic_variation = m_magnetic_variation;
    m_magnetic_variation = _magnetic_variation;

}

::ecore::EString GPS_NMEA0183_T_R::getFAA_Mode_Indicator() const
{
    return m_fAA_Mode_Indicator;
}

void GPS_NMEA0183_T_R::setFAA_Mode_Indicator(::ecore::EString _fAA_Mode_Indicator)
{

    ::ecore::EString _old_fAA_Mode_Indicator = m_fAA_Mode_Indicator;
    m_fAA_Mode_Indicator = _fAA_Mode_Indicator;

}

::ecore::EFloat GPS_NMEA0183_T_R::getHeading() const
{
    return m_heading;
}

void GPS_NMEA0183_T_R::setHeading(::ecore::EFloat _heading)
{

    ::ecore::EFloat _old_heading = m_heading;
    m_heading = _heading;

}

::ecore::EInt GPS_NMEA0183_T_R::getHeadingTrue() const
{
    return m_headingTrue;
}

void GPS_NMEA0183_T_R::setHeadingTrue(::ecore::EInt _headingTrue)
{

    ::ecore::EInt _old_headingTrue = m_headingTrue;
    m_headingTrue = _headingTrue;

}

// References

